app/dashboard/views.py

```diff
@@ -6495,3 +6496,63 @@ def file_upload(request):
         data = {'is_valid': False}
 
     return JsonResponse(data)
+
+@csrf_exempt
+def mautic_api(request, endpoint=''):
+    print(request.GET)
+    params = request.GET
+    credential = f"{settings.MAUTIC_USER}:{settings.MAUTIC_PASSWORD}"
+    token = base64.b64encode(credential.encode("utf-8")).decode("utf-8")
+    headers = {"Authorization": f"Basic {token}"}
+    print(headers)
+    print(endpoint)
+    print(request.body)
+    if request.body:
+        body_unicode = request.body.decode('utf-8')
+        payload = json.loads(body_unicode)
+        print(payload)
+
+    url = f"https://gitcoin-5fd8db9bd56c8.mautic.net/api/{endpoint}"
+    if request.method == 'GET':
+        response = requests.get(url=url, headers=headers, params=params).json()
+    elif request.method == 'POST':
+        response = requests.post(url=url, headers=headers, params=params, data=json.dumps(payload)).json()
+    elif request.method == 'PUT':
+        response = requests.put(url=url, headers=headers, params=params, data=json.dumps(payload)).json()
+    elif request.method == 'PUT':
+        response = requests.put(url=url, headers=headers, params=params, data=json.dumps(payload)).json()
+    elif request.method == 'PATCH':
+        response = requests.patch(url=url, headers=headers, params=params, data=json.dumps(payload)).json()
+
+    return JsonResponse(response)
+
+
+@csrf_exempt
+@require_POST
+def mautic_profile_save(request):
```

Comment:
We could add the @login_required decorator remove is_auth check

Reply:
I use to avoid @login_required for rest endpoints as will redirect user to github login instead of just silently failing on the console